services/sharedchannel/service.go

```diff
@@ -177,3 +178,16 @@ func (scs *Service) pause() {
 
 	scs.server.GetLogger().Debug("Shared Channel Service inactive")
 }
+
+// Makes the remote channel to be read only, only for guests.
+func (scs *Service) makeChannelReadOnly(channel *model.Channel) *model.AppError {
```

Comment:
Couldn't find the product requirements for this one anywhere, so will defer to @wiggin77 for this one. As is right now we define a remote channel as read-only, when guests cannot create a post to it(they can though react), also all members of that channel can post and react. There are essentially 4 layers of "readonlyness".

1) Guests cannot create a post
2) Guests cannot create a post and reactions
3) Guests and Members cannot create a post
4) Guests and Member cannot create a post and reactions(Essentially only admins can edit the channel)

Reply:
I thought the link in the ticket pointed directly to the right section but I see now it points to the section above it.  This is meant to be like an announcement channel, the second section in that link.  The use case from design meetings was sharing a Town Square like channel with other orgs.

Reply:
All done in https://github.com/mattermost/mattermost-server/pull/16803/commits/749ec630122e22d8f49487d7c88457cbc7ba70f4